imageregistry/v1/types.go

```diff
@@ -303,8 +302,32 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access level for the storage account.
+	// "Internal" means the storage account will be private, "External" means
+	// the storage account will be publicly accessible.
+	// Defaults to "External".
+	// +kubebuilder:validation:Enum="Internal";"External"
+	// +kubebuilder:default="External"
+	// +optional
+	NetworkAccess AzureNetworkAccessibility `json:"networkAccess,omitempty"`
+	// vnetName is the name of the vnet the registry operates in.
+	// +optional
+	VNetName string `json:"vnetName,omitempty"`
+	// subnetName is the name of the subnet the registry operates in.
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
+	// privateEndpointName is the name of the private endpoint for the registry.
+	// +optional
+	PrivateEndpointName string `json:"privateEndpointName,omitempty"`
 }
 
+type AzureNetworkAccessibility string
+
+const (
+	AzureNetworkAccessInternal AzureNetworkAccessibility = "Internal"
```

Comment:
Should you use Public vs. Private instead? Is Internal/External a dedicated keyword in Azure for marking an access as private/public?

Reply:
We can switch to Public/Private. I've chosen Internal/External because it reflects the `publish` key from the `install-config.yaml` (which will in the future be used to control network access for the registry storage account too)